cedar-lean/Cedar/Thm/Data/Set.lean

```diff
@@ -321,6 +353,74 @@ theorem inter_wf {α} [LT α] [StrictLT α] [DecidableLT α] [DecidableEq α] {s
   simp only [List.elem_eq_mem]
   exact h₃
 
+theorem inter_empty_left {α} [LT α] [StrictLT α] [DecidableLT α] [DecidableEq α] (s : Set α) :
+ Set.empty ∩ s = Set.empty
+:= by
+  cases s ; rename_i xs
+  simp only [Inter.inter, intersect, List.inter, elts, List.elem_eq_mem, empty, List.filter_nil]
+
+theorem inter_empty_right {α} [LT α] [StrictLT α] [DecidableLT α] [DecidableEq α] (s : Set α) :
+ s ∩ Set.empty = Set.empty
+:= by
+  cases s ; rename_i xs
+  simp only [Inter.inter, intersect, List.inter, elts, empty, List.elem_eq_mem, List.not_mem_nil,
+    decide_False, mk.injEq, List.filter_eq_nil, not_false_eq_true, implies_true]
+
+theorem inter_self_eq {α} [LT α] [StrictLT α] [DecidableLT α] [DecidableEq α] (s : Set α) :
+ s ∩ s = s
+:= by
+  simp only [Inter.inter, intersect, List.inter, elts, List.elem_eq_mem]
+  cases s ; rename_i xs
+  simp only [mk.injEq]
+  induction xs
+  case nil =>
+    simp only [List.not_mem_nil, decide_False, List.filter_nil]
+  case cons hd tl ih =>
+    simp only [List.mem_cons, true_or, decide_True, List.filter_cons_of_pos, List.cons.injEq, true_and]
+    rw [eq_comm]
+    rw (config := {occs := .pos [1]}) [← ih]
+    rw [List.filter_congr']
+    simp only [decide_eq_true_eq]
+    intro _ h
+    simp only [h, or_true]
+
+theorem intersects_def {α} [LT α] [StrictLT α] [DecidableLT α] [DecidableEq α] {s₁ s₂ : Set α} :
+ s₁.intersects s₂ = ¬ (s₁ ∩ s₂).isEmpty
+:= by
+  simp only [Bool.not_eq_true, eq_iff_iff]
+  simp only [intersects, List.any_eq_true, in_list_iff_in_set, contains_prop_bool_equiv]
+  constructor
+  case mp =>
+    intro h
+    replace ⟨x, h⟩ := h
+    by_contra hc
+    simp only [ne_eq, Bool.not_eq_false] at hc
+    simp only [empty_iff_not_exists, not_exists] at hc
+    specialize hc x
+    have _ := mem_inter_iff.mpr h
```

Comment:
```suggestion
    have := mem_inter_iff.mpr h
```

Reply:
Ah, I use `have _ := ` to make it clear that the introduced hypothesis will never be used by name.  Note that `have := ` introduces a usable name into the proof context (`this`).

Reply:
I see